### PR TITLE
qemu_img_convert_with_rate_limit: convert image with parameter -r

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1909752, 1909754, 1909757

### CODE REVIEW

**Comment by XueqiangWei** [comment]
depends on avocado-framework/avocado-vt/pull/2869

**Comment by XueqiangWei** [comment]
(01/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_0: PASS (425.78 s)
 (02/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_4611686018427387400: PASS (415.59 s)
 (03/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_9223372036854775295: PASS (423.76 s)
 (04/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0: PASS (450.73 s)
 (05/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400: PASS (458.60 s)
 (06/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295: PASS (459.54 s)
 (07/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_0: PASS (451.20 s)
 (08/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_4611686018427387400: PASS (439.33 s)
 (09/27) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_9223372036854775295: PASS (447.82 s)
 (10/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_0: PASS (234.90 s)
 (11/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_4611686018427387400: PASS (207.49 s)
 (12/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_9223372036854775295: PASS (208.36 s)
 (13/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0: PASS (219.44 s)
 (14/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400: PASS (210.33 s)
 (15/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295: PASS (218.73 s)
 (16/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_0: PASS (290.41 s)
 (17/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_4611686018427387400: PASS (290.76 s)
 (18/27) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_9223372036854775295: PASS (305.90 s)
 (19/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_0: PASS (344.09 s)
 (20/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_4611686018427387400: PASS (327.12 s)
 (21/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_9223372036854775295: PASS (321.88 s)
 (22/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0: PASS (339.06 s)
 (23/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400: PASS (338.20 s)
 (24/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295: PASS (281.66 s)
 (25/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_0: PASS (396.90 s)
 (26/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_4611686018427387400: PASS (418.23 s)
 (27/27) Host_RHEL.m8.u4.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_9223372036854775295: PASS (411.47 s)

 (1/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0.q35: PASS (576.02 s)
 (2/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400.q35: PASS (546.72 s)
 (3/3) Host_RHEL.m8.u4.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295.q35: PASS (503.36 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it. Many thanks.

File: qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    Windows:
+        guest_temp_file = "C:\testfile"
+        x86_64:
+            sync_bin = WIN_UTILS:\Sync\sync64.exe /accepteula
+        i386, i686:
+            sync_bin = WIN_UTILS:\Sync\sync.exe /accepteula
+    convert_source = ${images}
+    variants:
+        - rate_limit:
+            required_qemu = [5.2.0-1, )
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg:18]: move this to global ?  for this case is designed for rate limit, if version < 5.2.0-1, no need to run.

File: qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg [hunk 2]
```diff
@@ -0,0 +1,39 @@
+- qemu_img_convert_with_rate_limit:
+    virt_test_type = qemu
+    type = qemu_img_convert_with_rate_limit
+    start_vm = no
+    kill_vm = yes
+    force_create_image = no
+    guest_temp_file = "/var/tmp/convert.tmp"
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg:7]: just fyi.
/tmp/convert.tmp might be good, for we don't have to remove it post test.
But it's OK to use /var/tmp/xxx, cause the file is very small, and it is always convert.tmp created when running the tc.

Comment (zhencliu) [reply qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg:7]: just fyi.
Create a new file and check its md5sum to ensure a successful converting image, what about handle it in a simple way, start vm with converted image, and we can log into the vm.

File: qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg (hunk 1 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg:18]: Agree, I will update it. Many thanks.

File: qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg (hunk 2 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/qemu_img_convert_with_rate_limit.cfg]: Thanks zhenchao. I change to /tmp/convert.tmp. For the steps start vm with converted image and log into it to check file md5sum are contained in my test.

**Comment by XueqiangWei** [comment]
(1/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_0: PASS (215.30 s)
 (2/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_4611686018427387400: PASS (199.45 s)
 (3/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_raw.rate_limit.with_rate_limit_9223372036854775295: PASS (209.84 s)
 (4/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_0: PASS (230.66 s)
 (5/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_4611686018427387400: PASS (227.06 s)
 (6/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_qcow2.rate_limit.with_rate_limit_9223372036854775295: PASS (229.73 s)
 (7/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_0: PASS (279.99 s)
 (8/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_4611686018427387400: PASS (283.86 s)
 (9/9) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_convert_with_rate_limit.to_luks.rate_limit.with_rate_limit_9223372036854775295: PASS (286.44 s)

@zhencliu updated, please help review it again. Many thanks.

**Comment by zhencliu** [comment]
LGTM
Also, let's hold on till vt patch merged.

**Pull request merged**